ClickToCall.Web/Services/RequestValidationService.cs

```diff
@@ -10,16 +13,46 @@ public interface IRequestValidationService
 
     public class RequestValidationService : IRequestValidationService
     {
-        private readonly RequestValidator _requestValidator;
-
-        public RequestValidationService()
-        {
-            _requestValidator = new RequestValidator();
-        }
-
         public bool IsValidRequest(HttpContext context, string authToken)
         {
-            return _requestValidator.IsValidRequest(context, authToken);
+            // https://github.com/twilio/twilio-csharp/blob/ab0983ae00269746c5dbeb891955ae3edd83796d/src/Twilio.Twiml/RequestValidator.cs
+
+            if (context.Request.IsLocal)
+            {
+                return true;
+            }
+
+            // validate request
+            // http://www.twilio.com/docs/security-reliability/security
+            // Take the full URL of the request, from the protocol (http...) through the end of the query string (everything after the ?)
+            var value = new StringBuilder();
+            var fullUrl = context.Request.Url.AbsoluteUri;
+
+            value.Append(fullUrl);
+
+            // If the request is a POST, take all of the POST parameters and sort them alphabetically.
+            if (context.Request.HttpMethod == "POST")
+            {
+                // Iterate through that sorted list of POST parameters, and append the variable name and value (with no delimiters) to the end of the URL string
+                var sortedKeys = context.Request.Form.AllKeys.OrderBy(k => k, StringComparer.Ordinal).ToList();
```

Comment:
`Select` is the equivalent to a map function, it goes from something of type `a` to something to type `b`. In this case we're just sorting an array of strings, right?

Something interesting here is that `ToList` is completely unnecessary. `OrderedBy` will return an `IOrderedEnumerable` so you can use the `foreach` statement without additional transformations.